cloud/hack/test-integration.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The KubeEdge Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_ROOT=$(unset CDPATH && cd $(dirname "${BASH_SOURCE[0]}")/../../ && pwd)
+export KUBE_CONFIG=""
+export KUBE_APISERVER_URL="http://localhost:8080"
+
+runTests() {
+  cleanup
+
+  logStatus "Applying device crd yamls"
+  kubectl create -f "${SCRIPT_ROOT}/build/crds/devices/devices_v1alpha1_devicemodel.yaml"
+  kubectl create -f "${SCRIPT_ROOT}/build/crds/devices/devices_v1alpha1_device.yaml"
+
+  logStatus  "Running integration test cases"
+  export KUBE_RACE="-race"
+  make -C "${SCRIPT_ROOT}/cloud" test \
+      WHAT="${WHAT:-$(findIntegrationTestDirs | paste -sd' ' -)}" \
+      GOFLAGS="${GOFLAGS:-}" \
+      KUBE_TEST_ARGS="--alsologtostderr=true ${KUBE_TEST_ARGS:-} " \
+      KUBE_RACE="$KUBE_RACE" \
+
+  cleanup
+}
+
+findIntegrationTestDirs() {
+  (
+    cd "${SCRIPT_ROOT}/cloud"
+    find test/integration/ -name '*_test.go' -print0 \
+      | xargs -0n1 dirname | LC_ALL=C sort -u
+  )
+}
+
+cleanup() {
+  logStatus "Cleaning up"
+
+  local num_devices=$(kubectl get device | wc -l) && rc=$?
+  if [[ num_devices -gt 0 ]]; then
+    declare -a devices=("device-modbus-rtu" "device-modbus-tcp" "device-opcua" )
```

Comment:
Can we get the device directly from "kubectl get device"? so that the loop here might not be needed.

Reply:
the loop is for avoiding 3 separate lines for deleting every device. Without loop, we need to do the below : 
e.g
```sh
kubectl delete device device-modbus-rtu
kubectl delete device device-modbus-tcp
kubectl delete device device-opcua
```

Reply:
What I meant was there might be only one device, but it always loops three times to delete the potential not existing device, for example, `kubectl get device | awk -F' ' '{print $1}' | tail -2` will get all the devices that is actually created and then delete them.

Reply:
thanks @chendave for the inputs , I have made the suggested changes. Thanks.